### PR TITLE
Whisk/fix/void result check/188

### LINKED ISSUE #188: Check void functions result usage
## Code Example

```php
/** @return void */
function f() {}

$_ = f(); // Assigned void result.
```

## Actual Behavior

No warnings.

## Expected Behavior

Warning about void result usage.

Note: this is a harder part of #86, so we can implement it as independent patches.

### PR DESCRIPTION
**Pull request by whisk**
Should (partially) fix #188 

Added a check when trying to assign void result to a single variable.

### CODE REVIEW

**Comment by whisk** [comment]
Changed type inference on assignments slightly to it supports method calls.
Also added check for binary ops.

File: src/linttest/basic_test.go [hunk 1]
```diff
@@ -23,6 +23,72 @@ func TestCallStaticParent(t *testing.T) {
...
+	test.AddFile(`<?php
+	/**
+	* @return void
+	*/
+	function f() {}
+
+	f() & 1;
+	f() | 1;
+	f() ^ 1;
+	//f() && true; // skipped due to "undefined: Use true instead of true" report
```
Comment (quasilyte) [inline src/linttest/basic_test.go]: Related to #250.
For now, tests should declare builtin constants explicitly, like:
```
function define($_, $_) {}
define('false', 1 == 0);
define('true', 1 == 1);
```

Comment (whisk) [reply src/linttest/basic_test.go]: Should be fixed.

**Comment by quasilyte** [comment]
:tada:

**Pull request merged**